### PR TITLE
Fix: Compilation errors with DTO, OAuth and ManyToMany with User #11160

### LINKED ISSUE #11160: Compilation errors with DTO, OAuth and ManyToMany with User
##### **Overview of the issue**

Compilation errors below:

```
[ERROR] /Users/theo/Documents/perso/jhipster-many/src/test/java/com/mycompany/myapp/service/mapper/CountryMapperTest.java:[20,45] incompatible types: java.lang.String cannot be converted to java.lang.Long
[ERROR] /Users/theo/Documents/perso/jhipster-many/src/test/java/com/mycompany/myapp/service/dto/CountryDTOTest.java:[14,53] incompatible types: java.lang.String cannot be converted to java.lang.Long
[ERROR] /Users/theo/Documents/perso/jhipster-many/src/test/java/com/mycompany/myapp/service/dto/CountryDTOTest.java:[19,53] incompatible types: java.lang.String cannot be converted to java.lang.Long
```

##### **Motivation for or Use Case**

Project should compile.

##### **Reproduce the error**

Generate the app using the JDL below:

<details>
<summary>JDL file</summary>
<pre>
application {
  config {
    authenticationType oauth2
  }
  entities *
}

entity Country {
    name String
}

relationship ManyToMany {
    Country{User(login)} to User{Country(name)}
}

dto Country with mapstruct</pre>
</details>

##### **JHipster Version(s)**

```
jhipster@0.0.1-SNAPSHOT /Users/theo/Documents/perso/jhipster-many/mono
└── generator-jhipster@6.6.0

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "enableHibernateCache": true,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "jhiPrefix": "jhi",
    "languages": ["en", "fr"],
    "messageBroker": false,
    "nativeLanguage": "en",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "serviceDiscoveryType": false,
    "skipClient": false,
    "skipServer": false,
    "testFrameworks": [],
    "websocket": false,
    "authenticationType": "oauth2",
    "jhipsterVersion": "6.6.0",
    "baseName": "jhipster",
    "buildTool": "maven",
    "skipUserManagement": true,
    "clientPackageManager": "npm",
    "applicationType": "monolith",
    "cacheProvider": "ehcache",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "serverPort": "8080",
    "useSass": true,
    "embeddableLaunchScript": false,
    "creationTimestamp": 1579723742179,
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "blueprints": []
  },
  "entities": ["Country"]
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Country {
  name String
}
relationship ManyToMany {
  Country{user(login)} to User
}

dto Country with mapstruct
service Country with serviceClass

</pre>
</details>

##### **Environment and Tools**

openjdk version "12.0.2" 2019-07-16
OpenJDK Runtime Environment AdoptOpenJDK (build 12.0.2+10)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.15.1, JRE 12 Mac OS X amd64-64-Bit Compressed References 20190915_201 (JIT enabled, AOT enabled)
OpenJ9   - 0f66c6431
OMR      - ec782f26
JCL      - 9984ea0dd9 based on jdk-12.0.2+10)

git version 2.25.0

node: v10.18.1

npm: 6.13.4

yarn: 1.21.1

Docker version 19.03.5, build 633a0ea

docker-compose version 1.24.1, build 4667896b

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #11160

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by MathieuAA** [comment]
Hi! thanks for the PR. @Falydoor already dropped a PR for this (#11161) with a nice refactoring. Unless the first PR is merged, we'll merge this one instead.

File: generators/entity-server/templates/src/test/java/package/service/dto/EntityDTOTest.java.ejs [hunk 1]
```diff
@@ -22,9 +22,13 @@ import org.junit.jupiter.api.Test;
 import static org.assertj.core.api.Assertions.assertThat;
 import <%= packageName %>.web.rest.TestUtil;
 <%_
+let primaryKeyType = pkType;
 let hasOauthUser = false;
 for (idx in relationships) {
+    const otherEntityName = relationships[idx].otherEntityName;
+    const isUsingMapsId = relationships[idx].useJPADerivedIdentifier;
```
Comment (MathieuAA) [inline generators/entity-server/templates/src/test/java/package/service/dto/EntityDTOTest.java.ejs]: I would have named it `usingMapsId` instead, as it's a variable and not a function

Comment (qmonmert) [reply generators/entity-server/templates/src/test/java/package/service/dto/EntityDTOTest.java.ejs]: ok I update that

Comment (MathieuAA) [reply generators/entity-server/templates/src/test/java/package/service/dto/EntityDTOTest.java.ejs]: Thanks!

**Comment by qmonmert** [comment]
@MathieuAA PR ✅

**Comment by MathieuAA** [comment]
Closing this one, as the other PR has been merged. Thanks for your effort on this!

**Pull request closed without merging**